oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/time/ISODateTimeParser.java

```diff
@@ -31,60 +31,58 @@
    *  
    * @param isoDateTimeString ISO Date or ISO DateTime string
    * @return ZonedDateTime
-   * @throws Exception
+   * @throws Exception if Date or Date-Time pattern is not supported
    */
-  public static ZonedDateTime parseISODateTime(String isoDateTimeString) throws Exception{
+  public static ZonedDateTime parseISODateTime(String isoDateTimeString) throws Exception {
     ZonedDateTime zdt = null;
+
+    // always remove trailing Z, no other timezones are allowed anyways
```

Comment:
maybe "not allowed" is the wrong wording here: in a few places we actually [do](https://github.com/GIScience/oshdb/blob/0.5.9/oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/time/ISODateTimeParser.java#L20) [claim](https://docs.ohsome.org/ohsome-api/v1/time.html) [to](https://github.com/GIScience/oshdb/blob/master/documentation/manual/filters.md#timestamps) "support ISO 8601" timestamps. So, right now, the fact that currently only UTC strings are parsed is "just" a matter of us not bothering to implement these parts of the ISO standard. :wink: 

I see two possibilities now:
1. Document better that we only support a subset of the ISO standard, and explicitly define which one.
2. Embrace that the current implementation is still incomplete, but be open to have the code be extended further in the future (this might be done by switching to a different date-time parser library for full ISO support – I heard [Joda Time](https://www.joda.org/joda-time/) should be quite good).

Reply:
Extending the support for more timezones would result in a breaking change, because all query definitions without Z at the end would change their meaning and would/should be interpreted as local time. But which local time? The servers local time, but it should be interpreted as the users local time.
I think this would be a big source of errors. 
I would prefer a better documentation that we only support UTC and leave it to applications and users to convert local dates if they need them for some reasons.

Reply:
ok, that makes sense. so, let's start by documenting this in the javadoc above (https://github.com/GIScience/oshdb/blob/0.5.9/oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/time/ISODateTimeParser.java#L20)?

Reply:
Ok. Improved JavaDoc in this commit [4eb0dc6](https://github.com/GIScience/oshdb/pull/265/commits/4eb0dc6a18f0124d4964ba0b5b40f900e9a34380)

Reply:
I think it's also important to document more precisely in the ohsome-API.

Reply:
great, I've now also updated the oshdb documentation in this branch: https://github.com/GIScience/oshdb/blob/fix-issue-259-timestamp-formats/documentation/manual/filters.md#timestamps (https://github.com/GIScience/oshdb/pull/265/commits/ce15ccdd238e5e771e40261a81b1b0adc0794495)

Reply:
> I think it's also important to document more precisely in the ohsome-API.

yep -> I've opened https://github.com/GIScience/ohsome-api/issues/48 for that.